### PR TITLE
fix(meta): Meta node should be identified by its `meta_endpoint` parameter for cluster membership

### PR DESCRIPTION
**Pull request by jon-chuang**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
Meta node should be identified by its `meta_endpoint` parameter. Previously, this defaults to the `listen_addr`. The purpose of `meta_endpoint` is to serve as a unique identifier for cluster membership and for leadership election and connecting to leader.  

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### Types of user-facing changes

Possibly breaks any usage of risingwave that does not pass `meta_endpoint` when running with etcd.
### Release note
Meta node run with etcd must now specify `--meta-endpoint`

## Refer to a related PR or issue link (optional)
https://github.com/risingwavelabs/risingwave/issues/7526

### CODE REVIEW

**Review by shanicky — APPROVED** [review]
LGTM! 
Many thanks!

File: src/meta/src/lib.rs [hunk 1]
```diff
@@ -133,24 +132,30 @@ pub fn start(opts: MetaNodeOpts) -> Pin<Box<dyn Future<Output = ()> + Send>> {
...
+                        .collect(),
+                    credentials: match opts.etcd_auth {
+                        true => Some((opts.etcd_username, opts.etcd_password)),
+                        false => None,
+                    },
                 },
-            },
-            Backend::Mem => MetaStoreBackend::Mem,
+            ),
+            Backend::Mem => (
```
Comment (jon-chuang) [inline src/meta/src/lib.rs:154]: Should we still allow defaulting to listen_addr for no etcd? Anw, leader service should be useless in this case but...

**Comment by xxchan** [comment]
https://buildkite.com/risingwavelabs/docker/builds/13303#0185e9ad-6f6b-4368-83a9-e90788f2f948

😇
```
launching playground
error: Found argument '--host' which wasn't expected, or isn't valid in this context
	If you tried to supply `--host` as a value rather than a flag, use `-- --host`
USAGE:
    meta-node --listen-addr <LISTEN_ADDR>
```

**Comment by jon-chuang** [comment]
Oops, not caught by CI. Will be fixed in https://github.com/risingwavelabs/risingwave/pull/7530

**Comment by huangjw806** [comment]
This modification should not affect v0.1.16, because it is not included in it, and currently only affects nighty testing.

**Pull request merged**